int/api/cmd/execute_function.go

```diff
@@ -13,40 +11,37 @@ import (
 	"github.com/massalabs/thyra/pkg/wallet"
 )
 
-func NewExecuteFunction(walletStorage *sync.Map) *FunctionExecuter {
-	return &FunctionExecuter{walletStorage: walletStorage}
+func NewExecuteFunction() *FunctionExecuter {
+	return &FunctionExecuter{}
 }
 
-type FunctionExecuter struct {
-	walletStorage *sync.Map
-}
+type FunctionExecuter struct{}
 
 //nolint:nolintlint,ireturn
 func (f *FunctionExecuter) Handle(params operations.CmdExecuteFunctionParams) middleware.Responder {
 	addr, err := base58.CheckDecode((*params.Body.At)[1:])
 	if err != nil {
-		panic(err)
+		return operations.NewCmdExecuteFunctionUnprocessableEntity().WithPayload(
+			&models.Error{Code: errorCodeUnknownKeyID, Message: "Error: cannot decode Smart contract address : " + err.Error()})
 	}
 
 	addr = addr[1:]
 
-	value, exists := f.walletStorage.Load(*params.Body.Name)
-	if !exists {
+	wallet, err := wallet.Load(*params.Body.Nickname)
```

Comment:
params.Body.Nickname is a pointer because the field Nickname is not set as nullable:false and therefore the framework consider this fields as nullable.
This field should be mandatory, don't you think ?